clar2wasm/src/ast_visitor.rs

```diff
@@ -772,9 +772,12 @@ pub trait ASTVisitor {
                             args.get(1).unwrap_or(&DEFAULT_EXPR),
                             args.get(2).unwrap_or(&DEFAULT_EXPR),
                         ),
-                        BitwiseAnd | BitwiseOr | BitwiseNot | BitwiseXor2 => {
+                        BitwiseAnd | BitwiseOr | BitwiseXor2 => {
                             self.traverse_bitwise(builder, expr, native_function, args)
                         }
+                        BitwiseNot => {
+                            self.traverse_bitwise_not(builder, args.get(0).unwrap_or(&DEFAULT_EXPR))
+                        }
                         BitwiseLShift | BitwiseRShift => self.traverse_bit_shift(
                             builder,
                             expr,
```

Comment:
Why do we need another traverse function for not?
And actually, why did we need `traverse_bit_shift` to begin with?
@obycode why aren't all the bitwise functions grouped?

Reply:
Not is unary, the others are n-ary

Reply:
shift takes exactly 2 args

Reply:
If you look in traverse_arithmetic, you have functions that take two arguments (+, -, *, ...) and one argument (sqrti, log2)

Reply:
See above for my feelings on this.

Reply:
Well, it's really not such a big issue I guess, let's keep it like this then